packages/sdk/src/transactions/PSBTBuilder.ts

```diff
@@ -0,0 +1,475 @@
+import { networks, Psbt, Transaction } from "bitcoinjs-lib";
+import reverseBuffer from "buffer-reverse";
+
+import type { Network } from "../config/types";
+import {
+  INSTANT_BUY_SELLER_INPUT_INDEX,
+  MINIMUM_AMOUNT_IN_SATS,
+} from "../constants";
+import { FeeEstimator } from "../fee/FeeEstimator";
+import type { InputsToSign } from "../inscription/types";
+import { BaseDatasource } from "../modules/BaseDatasource";
+import { JsonRpcDatasource } from "../modules/JsonRpcDatasource";
+import {
+  convertSatoshisToBTC,
+  generateTxUniqueIdentifier,
+  getNetwork,
+  toXOnly,
+} from "../utils";
+import { InputType, processInput } from "./helper";
+import type { Output, UTXOLimited } from "./types";
+
+export interface PSBTBuilderOptions {
+  address: string;
+  changeAddress?: string;
+  feeRate: number;
+  network?: Network;
+  outputs: Output[];
+  publicKey: string;
+  autoAdjustment?: boolean;
+  instantTradeMode?: boolean;
```

Comment:
why is instant trade mode integrated into psbtbuilder?

Reply:
There are some instances where we have to bind the `sigHash` to the inputs e.g., in instant-trade flow. Also, in case of instant-trade flow, we need to skip the index where seller PSBT payloads are located.